src/OrchardCore.Modules/OrchardCore.Media/Filters/MediaFilters.cs

```diff
@@ -80,21 +81,25 @@ public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArgument
                 var quality = arguments["quality"];
                 var format = arguments["format"];
                 var anchor = arguments["anchor"];
+                var bgcolor = arguments["bgcolor"];
 
-                ApplyQueryStringParams(queryStringParams, width, height, mode, quality, format, anchor);
+                ApplyQueryStringParams(queryStringParams, width, height, mode, quality, format, anchor, bgcolor);
             }
             else
             {
                 queryStringParams = new Dictionary<string, string>();
 
-                var width = arguments["width"].Or(arguments.At(0));
-                var height = arguments["height"].Or(arguments.At(1));
-                var mode = arguments["mode"].Or(arguments.At(2));
-                var quality = arguments["quality"].Or(arguments.At(3));
-                var format = arguments["format"].Or(arguments.At(4));
-                var anchor = arguments["anchor"].Or(arguments.At(5));
+                var useNamed = arguments.Names.Any(); // Never mix named and indexed arguments as this leads to unpredictable results
```

Comment:
Great. I think this is a good choice.

Can you update the docs, just to say that if any arguments are named, all arguments must be named.

Should clear up any confusion.